tasks/generate-history/registerArchaeologists.ts

```diff
@@ -0,0 +1,65 @@
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import { BigNumber } from "ethers";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+import { generateHistoryConfig } from "./config";
+import { range } from "./helpers";
+
+/**
+ * Gives an archaeologist sarco token, approves the sarco token, and registers the archaeologist.
+ * Error stack traces are left out to keep the logs clean.
+ * @param hre The Hardhat Runtime Environment
+ */
+export async function registerArchaeologists(
+  hre: HardhatRuntimeEnvironment
+): Promise<SignerWithAddress[]> {
+  const accounts = await hre.ethers.getSigners();
+  const count = generateHistoryConfig.archaeologistCount;
+
+  if (count > accounts.length) {
+    throw new Error(`Not enough accounts. Count must be less than or equal to ${accounts.length}`);
+  }
+
+  // Get the contract
+  const diamond = await hre.ethers.getContract("Diamond_DiamondProxy");
+  const archaeologistFacet = await hre.ethers.getContractAt("ArchaeologistFacet", diamond.address);
+  const sarcoToken = await hre.ethers.getContract("SarcoTokenMock");
+
+  console.log("Registering archaeologists...");
+  const archaeologistSigners: SignerWithAddress[] = [];
+  for (let i = 0; i < count; i++) {
+    const account = accounts[i];
+    const minimumDiggingFee = 10;
+
+    // 1 week
+    const maximumRewrapInterval = 604800;
```

Comment:
Wondering if this should be randomized (or able to be customized), so that the front-end would display some hidden archaeologists as well as some non-hidden. Not sure the best way to accomplish this to ensure you get some of each bucket (shown + hidden based on maxRewrapInterval)

Reply:
We can. I started out that way but then realized that the rewrap interval is part of message that each arch signs, so the arch's criteria need to match up with what's being passed into the `createSarcophagus` function. So I set this as a constant to make things easier but we can certainly modify it.

Reply:
The original intent of this task was to create arch metrics, not to create archaeologists that can be viewed from the app. We can certainly add to this with a bit more work though.

Reply:
@knoll3 -- yeah, I think we can break this extra work out into a separate branch. 

Since the place the reputation stats are going to mainly be viewed is on the front-end archaeologist multi-select, it's probably worth figuring out how to tie these registered archs to the arch service. I'll create a project card for this.